### ISSUE TITLE
how to change the code making it train many epochs？

### ISSUE THREAD

**Issue by sunanlin13174** [CoinCheung/BiSeNet#227]
hi，thank you for your excellent work.  
I wanted to train more epochs to get better results, but found that this code only trained about one epoch? I simply added more epoch for circular training externally, but the training results are getting worse and worse, the loss is rising and the accuracy rate is declining. How can I adjust the parameters? Or change?

My training set has only 200 pictures, and the batchsize is 4.
![企业微信截图_16479142906049](https://user-images.githubusercontent.com/51011133/159392125-6d02fafe-f54f-48a3-ae51-09099fbbb57c.png)

![企业微信截图_16479142442476](https://user-images.githubusercontent.com/51011133/159392153-8b7a6481-01d2-4444-94bc-af7209437aad.png)

![企业微信截图_16479144841119](https://user-images.githubusercontent.com/51011133/159392366-875bc4f8-135e-46c7-9109-857a23e96a19.png)

**Comment by CoinCheung** [comment]
Sorry for being so late. I was pushed to do other work when I saw this and I forgot to reply :(

You do not need to change the code to assign epochs. You can simply compute the number of  iterations and set it in the configuration file.

For example, if your dataset has 1000 images and you train with a global batch size of 20, then one epoch means 50 iterations. Given this, if you want to train for 50 epoches, you can assign the number of iterations to be 10 * 50 = 500.

**Comment by coallar** [comment]
> 50

Hi,  if you want to train for 50 epoches, you can assign the number of iterations to be 10 * 50 = 500: should be changed to:
 50 *50 = 2500?

**Comment by CoinCheung** [comment]
Global batch size is `N * ims_per_gpu`.

**Comment by coallar** [comment]
> Global batch size is `N * ims_per_gpu`.

Got it, thanks

**Comment by dadaligoudan** [comment]
> Sorry for being so late. I was pushed to do other work when I saw this and I forgot to reply :(
> 
> You do not need to change the code to assign epochs. You can simply compute the number of iterations and set it in the configuration file.
> 
> For example, if your dataset has 1000 images and you train with a global batch size of 20, then one epoch means 50 iterations. Given this, if you want to train for 10 epoches, you can assign the number of iterations to be 10 * 50 = 500.

Hi, I met with the same problem. After training more epochs, the loss is getting bigger. I don't know how to solve this problem, could you please help me with it? Thanks a lot.

**Comment by CoinCheung** [comment]
@dadaligoudan Have you tried to adjust learning rate? Maybe a smaller learning rate would solve this problem?

**Comment by happymatlab** [comment]
I have 8000 images and batch_size=8, which means my total iterations are 1000. I wanted to train 80 epochs, and according to the above answer, I should set max_iter to 1000*80, which is 80,000, but the training still stopped at 1000. max_iter doesn't seem to work in the train loop..
